sssom/util.py

```diff
@@ -483,49 +486,38 @@ def _extract_enum_text(df: pd.DataFrame) -> pd.DataFrame:
 
 
 def merge_msdf(
-    msdf1: MappingSetDataFrame,
-    msdf2: MappingSetDataFrame,
+    *msdfs: MappingSetDataFrame,
     reconcile: bool = True,
 ) -> MappingSetDataFrame:
-    """Merge msdf2 into msdf1.
+    """Merge multiple MappingSetDataFrames into one.
 
-    :param msdf1: The primary MappingSetDataFrame
-    :param msdf2: The secondary MappingSetDataFrame
+    :param msdfs: A Tuple of MappingSetDataFrames to be merged
     :param reconcile: If reconcile=True, then dedupe(remove redundant lower confidence mappings)
         and reconcile (if msdf contains a higher confidence _negative_ mapping,
         then remove lower confidence positive one. If confidence is the same,
         prefer HumanCurated. If both HumanCurated, prefer negative mapping).
         Defaults to True.
     :returns: Merged MappingSetDataFrame.
     """
-    # Inject metadata of msdf into df
-    msdf1 = inject_metadata_into_df(msdf=msdf1)
-    msdf2 = inject_metadata_into_df(msdf=msdf2)
-
-    msdf1.df = _extract_enum_text(msdf1.df)
-    msdf2.df = _extract_enum_text(msdf2.df)
-
     merged_msdf = MappingSetDataFrame()
-    # If msdf2 has a DataFrame
-    if msdf1.df is not None and msdf2.df is not None:
-        # 'outer' join in pandas == FULL JOIN in SQL
-        merged_msdf.df = msdf1.df.merge(msdf2.df, how="outer")
-    else:
-        merged_msdf.df = msdf1.df
 
-    # merge the non DataFrame elements
-    merged_msdf.prefix_map = dict_merge(
-        source=msdf2.prefix_map,
-        target=msdf1.prefix_map,
-        dict_name="prefix_map",
+    # Inject metadata of msdf into df
+    msdf_with_meta = [inject_metadata_into_df(msdf) for msdf in msdfs]
+
+    # merge df [# 'outer' join in pandas == FULL JOIN in SQL]
+    df_merged = reduce(
+        lambda left, right: left.merge(
+            right,
+            how="outer",
+        ),
+        [_extract_enum_text(msdf.df) for msdf in msdf_with_meta],
```

Comment:
I dont understand exactly what these 7 lines of code do, but I assume its simply a iterative panda merge.. What does _extract_enum_text do?

Reply:
Yes, you're correct. It does an pandas merge of DataFrame objects. `_extract_enum_text` addresses the issue where we had enums in `MatchType` which gave an error e.g.: 
```
Unknown MatchTypeEnum enumeration code: [(text='Lexical', description='Lexical match'), (text='HumanCurated', description='Match based on human expert opinion')]
```

We are just concerned about the `text` portion of the object. Remember we faced this issue with the enum object that comes from LinkML?

Reply:
I remember, but its odd that the dataframe should contain these values - shouldn't this conversion be handled in `read_sssom_table(i)` ?